python_modules/dagster/dagster/_core/selector/subset_selector.py

```diff
@@ -96,20 +89,9 @@ class AssetSelectionData(
             pipeline snapshot lineage.
     """
 
-    def __new__(
-        cls,
-        asset_selection: Optional[AbstractSet[AssetKey]],
-        asset_check_selection: Optional[AbstractSet[AssetCheckKey]],
-        parent_job_def: "JobDefinition",
-    ):
-        from dagster._core.definitions.job_definition import JobDefinition
-
-        return super(AssetSelectionData, cls).__new__(
-            cls,
-            asset_selection=check.opt_set_param(asset_selection, "asset_selection", AssetKey),
-            asset_check_selection=asset_check_selection,
-            parent_job_def=check.inst_param(parent_job_def, "parent_job_def", JobDefinition),
-        )
+    asset_selection: AbstractSet[AssetKey]
+    asset_check_selection: Optional[AbstractSet[AssetCheckKey]]
+    parent_job_def: "JobDefinition"
```

Comment:
resolving these `ForwardRef`s  depend on being able to find the type for the string `"JobDefinition"` the first place this class is used at runtime, unless you do `ImportFrom["JobDefinition", "dagster._core.definitions.job_definition"]` which allows inserting a lazy import 

given the current callsite the implicit scheme should always work, so this is just a heads up comment

Reply:
Eenteresting. Going to try to move some stuff around to eliminate the forward refs if I have time.